pkg/store/instance.go

```diff
@@ -250,31 +264,96 @@ func PrintInstances(w io.Writer, instances []*Instance, format string) error {
 		}
 		fmt.Fprintln(w, string(b))
 	case "table":
+		types := map[string]int{}
+		archs := map[string]int{}
+		for _, instance := range instances {
+			types[instance.VMType]++
+			archs[instance.Arch]++
+		}
+		all := options != nil && options.AllFields
+		width := 0
+		if options != nil && options.IsTerminal {
+			width = options.TerminalWidth
+		}
+		hideType := false
+		hideArch := false
+		hideDir := false
+
+		columns := 1 // NAME
+		columns += 2 // STATUS
+		columns += 2 // SSH
+		if width != 0 && (columns+7)*8 > width && !all {
+			hideType = len(types) == 1
+		}
+		if !hideType {
+			columns++ // VMTYPE
+		}
+		hideArch = len(archs) == 1 && !all
+		if !hideArch {
+			columns++ // ARCH
+		}
+		columns++ // CPUS
+		columns++ // MEMORY
+		columns++ // DISK
+		if width != 0 && (columns+2)*8 > width && !all {
+			hideDir = true
+		}
+		if !hideDir {
+			columns += 2 // DIR
+		}
+		_ = columns
+
 		w := tabwriter.NewWriter(w, 4, 8, 4, ' ', 0)
-		fmt.Fprintln(w, "NAME\tSTATUS\tSSH\tVMTYPE\tARCH\tCPUS\tMEMORY\tDISK\tDIR")
+		fmt.Fprint(w, "NAME\tSTATUS\tSSH")
+		if !hideType {
+			fmt.Fprint(w, "\tVMTYPE")
+		}
+		if !hideArch {
+			fmt.Fprint(w, "\tARCH")
+		}
+		fmt.Fprint(w, "\tCPUS\tMEMORY\tDISK")
+		if !hideDir {
+			fmt.Fprint(w, "\tDIR")
+		}
+		fmt.Fprintln(w)
 
-		u, err := user.Current()
+		homeDir, err := homeDir()
 		if err != nil {
 			return err
 		}
-		homeDir := u.HomeDir
```

Comment:
I don't understand the point of this change. The `homeDir()` wrapper seems to add complexity for no benefit.

Reply:
It was needed for the unit test, but I can duplicate it there instead.